extensions/api/auth-spi/src/main/java/org/eclipse/dataspaceconnector/api/auth/AuthenticationRequestFilter.java

```diff
@@ -29,7 +29,7 @@ public AuthenticationRequestFilter(AuthenticationService authenticationService)
     }
 
     @Override
-    public void filter(ContainerRequestContext requestContext) {
+    public void filter(ContainerRequestContext requestContext) throws NotAuthorizedException {
```

Comment:
add comment explaining why RuntimeException is explicitly declared?

and explain this in the ADR maybe?

Reply:
Is it correct that the reason is to have `api-core` as an API dependency in `auth-spi`? Not sure how maintainable this is, I think developers are not used to declare RuntimeExceptions and there is no failing check at the moment that would verify this.

Reply:
Removed Exception from signature and marked it with `@ExcludeFromDependecyAnalysisReport`